### PR TITLE
UCX: set memtype in regmr and send recv

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
Specify memory type in regmr,send,recv functions to avoid wrong detection if cuda libs linked statically

### CODE REVIEW

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>FAIL</b> (click for details)</summary>

**Note:** the artefacts will be deleted after 06-Aug-2020

Artefact file | Status
-- | --
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/dd3a114437ea9f897f65854ba9ca8c19/raw/ff1da1064dbb61052e88236720b60a7abf28e8fb/configure_sharp_startup.log) | :x: FAIL
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/dd3a114437ea9f897f65854ba9ca8c19/raw/d0b6bf5315a098e419e5a9f61c9d754fa7be4487/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by bureddy** [comment]
bot:mlx:retest

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 07-Aug-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/e903b6fdd95cd125f040004b9ac1da26/raw/3e8b6812ad3df9f1a2d53d6c5c103b36b67750bd/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/e903b6fdd95cd125f040004b9ac1da26/raw/d4bdc174984c2a2ba0bae3a7251cf4b9ceb8bbfa/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/e903b6fdd95cd125f040004b9ac1da26/raw/5224080d72c9f56b06f83e3d0d017c387e269198/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/e903b6fdd95cd125f040004b9ac1da26/raw/4c21e36f1959bf2f1d8575f69a0cb09e1f9be188/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: src/ucx_plugin.c [hunk 1]
```diff
@@ -463,18 +484,30 @@ ncclResult_t nccl_ucx_regmr(void* comm, void* data, int size, int type, void** m
...
+                            UCP_MEM_MAP_PARAM_FIELD_LENGTH |
+                            UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE; 
+  mmap_params.memory_type = mh->mem_type;
+#else
+  mh->mem_type = type;
+  mmap_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                           UCP_MEM_MAP_PARAM_FIELD_LENGTH; 
+#endif
+  mmap_params.address    = (void*)reg_addr;
+  mmap_params.length     = reg_size;
```
[the remark this answers is not in the document]
Comment (Sergei-Lebedev) [reply src/ucx_plugin.c:508]: fixed in #51

**Pull request merged**